### ISSUE TITLE
"Nonce too low" error on 4.10

### ISSUE THREAD

**Issue by bdresser** [MetaMask/metamask-extension#5336]
Multiple users reported the below error on 4.10 via support. Possibly showing incorrectly. No longer an issue on 4.11 but should be investigated before we proceed to 4.12

> I checked the nonce of my latest transaction on Etherscan, it is 1945. But my metamask uses 1912 which is much smaller than it is expected

![image](https://user-images.githubusercontent.com/1016190/45974399-0fa2fd00-bff6-11e8-94a0-38f1677116a0.png)

Full error:
`
ALERT: [ethjs-rpc] rpc error with payload {"id":2471053487490,"jsonrpc":"2.0","params":["0xf8ab82384a84ee6b280083013b00949d8be94d0612170ce533ac4d7b43cc3cd91e5a1a80b844a9059cbb000000000000000000000000634bdf6f8f26550e54dd4245179100970b05b1ba00000000000000000000000000000000000000000000071ceae4788d7b18000025a0890c811b23971f4fd384e3300d9e916a23ecb685fe9f277beff5615fb8351c9ca020f3bf7570362c5adaa616390e89d2da76fc8623f25e2ae242d2737d71afa197"],"method":"eth_sendRawTransaction"} {"code":-32000,"message":"nonce too low”}
`

**Comment by danfinlay** [comment]
I believe this was fixed by #5310.

**Comment by Juan-cc** [comment]
Same happening here guys. I am using MetaMask Version 4.12.0

ALERT: [ethjs-rpc] rpc error with payload {"id":5410027058364,"jsonrpc":"2.0","params":["0xf9014902843b9aca0...51500000..e...1a9d507b6e"],"method":"eth_sendRawTransaction"} {"code":-32000,"message":"nonce too low"}
CANCEL
CONFIRM

Any idea how to prevent this from happening?

**Comment by Juan-cc** [comment]
I found the solution ([here](https://github.com/MetaMask/metamask-extension/issues/1569)). I had to restart metamask going to mainnet and back to rinkeby.

**Comment by slav86** [comment]
Hi guys, I have the same problem now. I had a few transactions pending and one with nonce 22 and 23 went through but there is none 21 that is failing due to error "nonce too low". MetaMask only allows me to resubmit nonce 21 now, as before it would only allow 22 and 23. 
I tried changing networks several times with no result.

Any other option?

**Comment by tabcat** [comment]
still getting this error when connected to private network, tried fix in https://github.com/MetaMask/metamask-extension/issues/1569, 
using v4.12

**Comment by tabcat** [comment]
@jennypollack i believe i read the fix wrong initially, as I never reset the account until it was suggested by a person in the metamask gitter channel. i have and am using the reset account to unblock my accounts now.

**Comment by gorgos** [comment]
Nothing worked for me, I was using an account from the Nano Ledger S. The only thing that finally did work was reinstalling MetaMask.

**Comment by bdresser** [comment]
closing as we're now well past 4.10. let me know if you're still experiencing this issue.

**Comment by bneiluj** [comment]
Still having the issue with Metamask and Ledger
<img width="336" alt="Screen Shot 2019-04-20 at 09 39 31" src="https://user-images.githubusercontent.com/3680497/56454399-3a68f000-6350-11e9-8f3b-5dc7eac5a44c.png">
MetaMask Version 6.3.2, Chrome and Brave

**Comment by brunobar79** [comment]
@bneiluj  Can you check which version of the ledger firmware do you have?
Also the version of the ETH app which can check via the ledger live app.

We are aware of some issues earlier versions  of both so please make sure you have at least:

Firmware version v1.5.5+

**Comment by MmeKarl** [comment]
Hi I'm having the same issue. Everything is updated. I tried everything. Nothing worked. I've got some eth stock on idex that I'm unable to move and I'm having the same message when I try to transfer zil.

**Comment by bdresser** [comment]
@MmeKarl have you tried resetting your account from Settings > Advanced?

**Comment by bneiluj** [comment]
I've tried: 
1. resetting metamask account from Settings > Advanced
2. restarting metamask
3. restarting ledger 

Ledger Nano S
Firmware version 1.5.5
Ethereum
Version 1.2.4
MetaMask Version
6.4.1

**Comment by frankiebee** [comment]
@bahnju could you please email your state logs to support@metamask.io and reference this issue number?

**Comment by frankiebee** [comment]
as well as anyone else in this thread still having this issue ^

**Comment by bneiluj** [comment]
After running through the this bug over and over it seems that NOW (maybe after all updates) the only way to fix this issue is to remove Metamask from browser and install it again. I don't know why it didn't work before.  I'll try to send state logs ASAP.

**Comment by livnev** [comment]
I had the same issue, using a Ledger Nano S. The only thing that helped was deleting and reinstalling Metamask.

**Comment by tmashuang** [comment]
Closing due to inactivity, if anyone here is still experiencing this open a new issue with proper reproduction steps.
